sdk/src/main/java/com/mapbox/maps/renderer/gl/GlUtils.kt

```diff
@@ -0,0 +1,54 @@
+package com.mapbox.maps.renderer.gl
+
+import android.opengl.GLES20
+import android.opengl.Matrix
+import com.mapbox.maps.BuildConfig
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.nio.FloatBuffer
+
+object GlUtils {
+
+  fun FloatArray.toFloatBuffer(): FloatBuffer = ByteBuffer.allocateDirect(size * 4)
+    .order(ByteOrder.nativeOrder())
+    .asFloatBuffer().also {
+      it.put(this@toFloatBuffer)
+      it.rewind()
+    }
+
+  /**
+   * @param type GLES20.GL_VERTEX_SHADER or GLES20.GL_FRAGMENT_SHADER
+   */
+  fun loadShader(type: Int, shaderCode: String) =
+    GLES20.glCreateShader(type).also { shader ->
+      GLES20.glShaderSource(shader, shaderCode)
+      GLES20.glCompileShader(shader)
+    }
+
+  fun checkError(cmd: String? = null) {
+    if (BuildConfig.DEBUG) {
+      when (val error = GLES20.glGetError()) {
+        GLES20.GL_NO_ERROR -> {}
+        GLES20.GL_INVALID_ENUM -> throw RuntimeException("$cmd -> error in gl: GL_INVALID_ENUM")
+        GLES20.GL_INVALID_VALUE -> throw RuntimeException("$cmd -> error in gl: GL_INVALID_VALUE")
+        GLES20.GL_INVALID_OPERATION -> throw RuntimeException("$cmd -> error in gl: GL_INVALID_OPERATION")
+        GLES20.GL_INVALID_FRAMEBUFFER_OPERATION -> throw RuntimeException("$cmd -> error in gl: GL_INVALID_FRAMEBUFFER_OPERATION")
+        GLES20.GL_OUT_OF_MEMORY -> throw RuntimeException("$cmd -> error in gl: GL_OUT_OF_MEMORY")
+        else -> throw RuntimeException("$cmd -> error in gl: $error")
+      }
+    }
+  }
+
+  fun checkCompileStatus(shader: Int) {
```

Comment:
If this method only for **debug** purpose only should we mark it somehow?

Reply:
Not sure, do you have a suggestions? I could add a comment but in general method seem obvious without it